src/components/hack-details.js

```diff
@@ -1,100 +1,174 @@
 import React, { useState, useEffect } from "react";
 import Axios from "axios";
+import Team from "./subcomponents/team";
 
 // UI imports
-import { Grid, Typography, Paper, Button, Chip } from "@material-ui/core";
+import {
+  Grid,
+  Typography,
+  Paper,
+  Button,
+  Chip,
+  CircularProgress
+} from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
 import ErrorOutlineOutlinedIcon from "@material-ui/icons/ErrorOutlineOutlined";
 
+import { connect } from "react-redux";
+
 const useStyles = makeStyles(theme => ({
   root: {
     margin: theme.spacing(2),
     padding: theme.spacing(2)
   },
   rightFeild: {
     textAlign: "right"
+  },
+  loading: {
+    marginTop: theme.spacing(10)
+  },
+  chipMArgin: {
+    marginRight: theme.spacing(1)
   }
 }));
 
-function HackDetails(props) {
+export function HackDetails(props) {
   const classes = useStyles();
 
   const [data] = useState();
   const [title, setTitle] = useState("");
   const [description, setDescription] = useState("");
+  const [team, setTeam] = useState();
   const [goal, setGoal] = useState("");
 
   useEffect(() => {
     //console.log(props.match.params);
+
     if (props.match !== undefined) {
-      Axios.get(process.env.REACT_APP_API_URL+"hackdetail/" + props.match.params.id)
-        .then(res => {
-          setTitle(res.data.title);
-          setDescription(res.data.description);
-          setGoal(res.data.goal);
-        })
-        .then(() => {
-          console.log(data);
-        });
-    } else {
-      setTitle("Hack-Zero");
-      setDescription(
-        "This is where the description/idea for the hack will be displayed when there is one. Lorem ipsum dolor sit amet, consectetur adipiscing elit. In faucibus in tortor a molestie. Vestibulum congue, eros et ultricies vehicula, sapien est sollicitudin risus, ut volutpat augue tellus in neque. Aliquam erat volutpat."
-      );
-      setGoal("To win and get that dosh but also the experience");
+      Axios.get(
+        process.env.REACT_APP_API_URL + "hackdetail/" + props.match.params.id
+      ).then(res => {
+        setTitle(res.data.title);
+        setDescription(res.data.description);
+        setGoal(res.data.goal);
+        setTeam(res.data.team);
```

Comment:
question: should we have 4 different useState for these different parts? or just 1 for `setHack(res.data)`

there is no right or wrong answer, but given these values always set together, and if we wanted to extend the page with more information later, it might make more sense to combine them into one item in state to reduce the effort later.

Reply:
combining them does make the code a little more readable, I have made the changes